jade/errors.py

```diff
@@ -1,4 +1,162 @@
+import sys
+import traceback
 
 
-class TrustedClientVerificationError(RuntimeError):
-    pass
+class RequestError(RuntimeError):
+    "An error occured inside of JADE while processing a request"
+    TYPE = None
+    SUBTYPE = None
+    HTTP_CODE = 500
+
+    def format_detail(self):
+        return None
+
+    def format_json(self):
+        return {'type': self.TYPE, 'subtype': self.SUBTYPE,
+                'message': self.__class__.__doc__,
+                'detail': self.format_detail()}
+
+
+class UnknownError(RequestError):
+    "An unknown error occured inside of JADE while processing a request."
+    TYPE = "unknown"
+
+    def __init__(self, e):
+        super().__init__()
+        self.e = e
+
+    def format_detail(self):
+        doc = {'exception': str(self.e)}
+        if __debug__:
+            doc['traceback'] = get_tb()
+        return doc
+
+
+###############################################################################
```

Comment:
I guess this #### should be removed :D

Reply:
It's nice for visual separation of different groups of exceptions.

Reply:
That could be done by supplying the groups of exceptions in different modules, then including each module here.